scripts/mysql/upgrade-0.5.0-to-0.6.0-mysql.sql

```diff
@@ -0,0 +1,92 @@
+--
+-- Copyright 2024 Datastrato Pvt Ltd.
+-- This software is licensed under the Apache License version 2.
+--
+
+-- <issue-3099: Store role, user, and group under {metalake}.system.{role|user|group} in relation storage>
+ALTER TABLE `user_meta`
+    ADD COLUMN `catalog_id` BIGINT(20) UNSIGNED NOT NULL COMMENT 'catalog id' AFTER `metalake_id`;
+ALTER TABLE `user_meta`
+    ADD COLUMN `schema_id` BIGINT(20) UNSIGNED NOT NULL COMMENT 'schema id' AFTER `catalog_id`;
+
+ALTER TABLE `group_meta`
+    ADD COLUMN `catalog_id` BIGINT(20) UNSIGNED NOT NULL COMMENT 'catalog id' AFTER `metalake_id`;
+ALTER TABLE `group_meta`
+    ADD COLUMN `schema_id` BIGINT(20) UNSIGNED NOT NULL COMMENT 'schema id' AFTER `catalog_id`;
+
+ALTER TABLE `role_meta`
+    ADD COLUMN `catalog_id` BIGINT(20) UNSIGNED NOT NULL COMMENT 'catalog id' AFTER `metalake_id`;
+ALTER TABLE `role_meta`
+    ADD COLUMN `schema_id` BIGINT(20) UNSIGNED NOT NULL COMMENT 'schema id' AFTER `catalog_id`;
+
+-- This stored procedure is designed to initialize system catalog and schema for each metalake
+-- and it updates the `catalog_id` and `schema_id` in the `user_meta`, `group_meta`, and `role_meta` tables to the newly generated `catalog_id` and `schema_id`.
+DELIMITER $$
+
+CREATE PROCEDURE InitializeSystemCatalogAndSchema()
```

Comment:
@yuqi1129 I use the procedure to initialize system catalog and schema for each metalake, do you have any suggestion about this

Reply:
Should we maintain this logic for SQL or code?

Reply:
I think this is a one-time job, using an SQL file to complete the database upgrade is more convenient for users.

Reply:
I have no incline about this point as it seems to be more convenient to use a single file to update the SQL. one question:  

are these procedure sentences compatible with MySQL 5.6, 5.7, and 8.x at the same time?

Reply:
I have tested it in MySQL 5.7, and I will test it in 5.6 and 8.0 later.

Reply:
@yuqi1129 I hava test it in MySQL 5.6 and 8.0, it's work

using MySQL 8.0 as an example
```
docker run --rm --name mysql8.0 -e MYSQL_ROOT_PASSWORD=lwyang -d mysql:8.0
-- init databse and insert metalakes

docker exec -i mysql8.0  mysql -plwyang -Dgravitino < /home/xm/gravitino/mysql-0.5.0.sql
docker exec -i mysql8.0  mysql -plwyang -Dgravitino < /home/xm/gravitino/upgrade.sql
```



in MySQL 5.6 and 8.0, i insert into 3 metalakes
<img width="1429" alt="image" src="https://github.com/datastrato/gravitino/assets/23456873/cf5f54ec-9aaa-4c8e-91c8-54c679bfdf9d">

then execute upgrade sql
<img width="692" alt="image" src="https://github.com/datastrato/gravitino/assets/23456873/a540a0db-1b49-4610-8025-464f3ef29629">

the catalog and schema data is initialized
![image](https://github.com/datastrato/gravitino/assets/23456873/17e603a7-7b7f-4d8f-8bec-77e6acbe3023)
![image](https://github.com/datastrato/gravitino/assets/23456873/28326581-b884-4aa8-8c61-287a2f1bf756)

Reply:
@jerryshao 
Please help to check if using the procedure is a good idea? Thank you.